src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -112,15 +118,17 @@ readyState switch
 
         internal async Task ConnectAsyncJavaScript(Uri uri, CancellationToken cancellationToken, List<string>? requestedSubProtocols)
         {
-            // Check that we have not started already
-            int priorState = Interlocked.CompareExchange(ref _state, (int)InternalState.Connecting, (int)InternalState.Created);
-            if (priorState == (int)InternalState.Disposed)
+            // Check that we have not started already.
+            switch ((InternalState)Interlocked.CompareExchange(ref _state, (int)InternalState.Connecting, (int)InternalState.Created))
```

Comment:
I may have asked this before, but... why is CompareExchange being used?  I thought this component was entirely single threaded?

Reply:
I actually had this in another branch and PR but ran out of time.  Went ahead and addressed this here.